### PR TITLE
[RW-9991][risk=no] Move delete disk business logic from controller to service.

### PR DESCRIPTION
**Pull request by evrii**
Moved delete disk business logic from controller to service. Hoping to reuse some of this when I add functionality to delete disk as an admin (in a future PR).

Ran disk controller and service tests locally without issue.

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/resources/workbench-api.yaml [hunk 1]
```diff
@@ -4267,7 +4267,7 @@ paths:
           schema:
             "$ref": "#/definitions/ListDisksResponse"
         404:
-          description: Workspace found
+          description: Workspace not found
```
Comment (evrii) [inline api/src/main/resources/workbench-api.yaml:4270]: Language issue from previous PR.

File: api/src/test/java/org/pmiops/workbench/disk/DiskServiceTest.java [hunk 2]
```diff
@@ -92,4 +96,34 @@ public void test_findByWorkspaceNamespace() {
...
+    when(mockWorkspaceService.lookupWorkspaceByNamespace(WORKSPACE_NS)).thenReturn(workspace);
+    diskService.deleteDisk(WORKSPACE_NS, diskName);
+    verify(mockLeonardoApiClient).deletePersistentDisk(GOOGLE_PROJECT_ID, diskName);
+  }
+
+  @Test
+  public void deleteDisk_notFound() {
+    String diskName = user.generatePDName();
+
+    when(mockWorkspaceService.lookupWorkspaceByNamespace(WORKSPACE_NS))
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/disk/DiskServiceTest.java:113]: this is also testing workspace-not-found

**Review by jmthibault79 — APPROVED** [review]
Nice small improvement to our structure.  Tested locally, delete-disk continues to work.

File: api/src/main/java/org/pmiops/workbench/api/DisksController.java [hunk 3]
```diff
@@ -20,15 +21,18 @@
 public class DisksController implements DisksApiDelegate {
   private static final Logger log = Logger.getLogger(DisksController.class.getName());
 
+  private final DiskService diskService;
```
Comment (dmohs) [inline api/src/main/java/org/pmiops/workbench/api/DisksController.java:24]: I couldn't get this to work when I tried it, but I wish we could just annotate these and then not have to add the boilerplate to the constructor.

File: api/src/main/resources/workbench-api.yaml (hunk 1 above)
Comment (dmohs) [reply api/src/main/resources/workbench-api.yaml:4270]: Not a change suggestion, but my instinct would be to leave these descriptions out unless "Not Found" (the meaning of 404) doesn't carry enough information. I imagine it does in most cases.

**Review by dmohs — APPROVED** [review]
LGTM.

File: api/src/main/resources/workbench-api.yaml (hunk 1 above)
Comment (evrii) [reply api/src/main/resources/workbench-api.yaml:4270]: I'm wondering if 404 is appropriate here. @yonghaoy mentioned this in a previous PR https://github.com/all-of-us/workbench/pull/7678#discussion_r1194255484 The 404 is thrown when the workspace of the disk that you are trying to delete does not exist. It is not saying that the disk that you are trying to delete does not exist.

I would love to get your thoughts on a better approach here.

I am considering something like this in order to throw a more understandable error.

```
  public void deleteDisk(String workspaceNamespace, String diskName) {
    String  googleProject;
    try {
      googleProject =
          workspaceService.lookupWorkspaceByNamespace(workspaceNamespace).getGoogleProject();
    } catch(NotFoundException exception){
      throw new ServerErrorException("Workspace does not exist");
    }
    leonardoNotebooksClient.deletePersistentDisk(googleProject, diskName);
  }
```

**Pull request merged**